cmd/snap-bootstrap/partition/sfdisk.go

```diff
@@ -149,24 +145,26 @@ func positionedVolumeFromDump(dump *sfdiskDeviceDump) (*gadget.LaidOutVolume, er
 			Type:       p.Type,
 			Filesystem: bd.FSType,
 		}
-		ps[i] = gadget.LaidOutStructure{
-			VolumeStructure: &structure[i],
-			StartOffset:     gadget.Size(p.Start) * sectorSize,
-			Index:           i + 1,
+		ds[i] = DeviceStructure{
+			LaidOutStructure: gadget.LaidOutStructure{
+				VolumeStructure: &structure[i],
+				StartOffset:     gadget.Size(p.Start) * sectorSize,
+				Index:           i + 1,
+			},
+			Node: p.Node,
 		}
 	}
 
-	pv := &gadget.LaidOutVolume{
-		Volume: &gadget.Volume{
-			ID:        ptable.ID,
-			Structure: structure,
-		},
-		Size:             gadget.Size(ptable.LastLBA),
-		SectorSize:       sectorSize,
-		LaidOutStructure: ps,
+	dl := &DeviceLayout{
```

Comment:
Added `Label`, more fields can be added along the way as the need arises.

Reply:
From the sfdisk documentation (and source code), it seems that the MBR label is `dos` and not `msdos`, do you confirm that?

Reply:
Yes, sorry for that. Parted uses `msdos` :smile:

Reply:
How should we act when the gadget partition schema is `mbr,gpt`? Does it indicate a hybrid MBR partitioning or just a simple protective MBR? It seems that in such case sfdisk will show it as GPT so I'll verify it this way, but please correct me if a different action should be taken.

Reply:
That's a good question. Right now, we ignored `mbr,gpt` hybrid and error out if schema is not either `mbr` or `gpt`. This is slightly inconsistent, because we still allow the structy `type` to be a hybrid `<mbr-0x-style-type>,<uuid>`. However, we did not have any complaints yet, but it'd be nice to fill the gap.

If sfdisk is lacking this information in the dump output, we could work out ourselves whether an MBR is in use by inspecting the boot signature, IIRC the last 2 bytes of the first sector.

In this PR it's probably best to leave a TODO. In follow ups, we could first add detection and error out, then implement support and checks for `mbr,gpt` the `gadget` package, lastly update the code here to fill the schema correctly when a hybrid setup is detected.

Reply:
Thanks, I'll add a comment and leave it unhandled for now.

Reply:
I've added support for hybrid schemas in the gadget package in #7721 now.